include/engine/geospatial_query.hpp

```diff
@@ -348,6 +348,56 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
                 coordinates->at(data.u), coordinates->at(data.v), input_coordinate,
                 point_on_segment, ratio);
 
+        const auto translateAngleToLocation = [](const double angle)
```

Comment:
No need to go over all this angle calculations. We can use what is already there. This can be simplified to something like:

```cpp
if (ratio < std::numeric_limits<double>::epsilon())
   return RelativeLocation::BEHIND;

if (ratio-1. < std::numeric_limits<double>::epsilon())
   return RelativeLocation::AHEAD;

auto distance = coordinate_caluclation::havensineDistance(point_on_segment, input_coordinate);
if (distance > 100. || distance < 3.)
    return RelativeLocation::INVALID;

bool is_left_of = coordinate_calculatetion::is_left_of(coordinates[data.u], coordinates[data.v], coordinate);

if (is_left_of)
  return RelativeLocation::Left;

return RelativeLocation::Right;
```
You can port `is_left_of` over from [here](https://github.com/TheMarex/deberg/blob/master/geometry.hpp#L135-L148). Coordinates need to be transformed to mercator first in the helper function.